### ISSUE TITLE
WrongType & KeyNotFound Errors

### ISSUE THREAD

**Issue by ZonerDan** [Powerspace/pg2bq#4]
I have modified my configuration file in multiple ways and I continue to get one of the following errors regardless of the changes I make:

When using JSON format:

[main] ERROR com.powerspace.pg2bq.Main$ - Incorrect configuration: KeyNotFound(jdbc,None,Set("jdbc")), KeyNotFound(gcloud,None,Set("gcloud"))

When using HOCON format that matches the example on your README:

[main] ERROR com.powerspace.pg2bq.Main$ - Incorrect configuration: WrongType(STRING,Set(OBJECT),None,jdbc), WrongType(STRING,Set(OBJECT),None,gcloud)

**Comment by sderosiaux** [comment]
Hi @ZonerDan,

Can you paste the content of your configuration file?
It seems you're passing a string to `jdbc` instead of an object `jdbc { ... }`, same for `gcloud`. Or some odd character is present?

**Comment by ZonerDan** [comment]
**JSON Version**:

{
"jdbc": {
"url": "<my_jdbc_connection_string>",
"user": "<my_user>",
"password": "",
"tables": ["people"]
},
"gcloud": {
"project": "<my_project_id>",
"service-account-key-path": "<path_to_my_json_file>",
"bq.dataset": "<my_dataset>",
"gcs.tmp-bucket": "pg-export"
}
}

**HOCON Version**:

jdbc {
url = "jdbc:postgresql://q-liferay-app01.qa.autozone.com:5433"
user = "<my_user>"
password = ""
tables = ["people"]
}

gcloud {
project = "<my_project_id>"
service-account-key-path = "<path_to_my_json_file>"
bq.dataset = "<my_dataset>"
gcs.tmp-bucket = "pg-export"
}

**Comment by ZonerDan** [comment]
**JSON Version**:

{
"jdbc": {
"url": "<my_jdbc_connection_string>",
"user": "<my_user>",
"password": "",
"tables": ["people"]
},
"gcloud": {
"project": "<my_project_id>",
"service-account-key-path": "<path_to_my_json_file>",
"bq.dataset": "<my_dataset>",
"gcs.tmp-bucket": "pg-export"
}
}

**HOCON Version**:

jdbc {
url = "<my_jdbc_connection_string>"
user = "<my_user>"
password = ""
tables = ["people"]
}

gcloud {
project = "<my_project_id>"
service-account-key-path = "<path_to_my_json_file>"
bq.dataset = "<my_dataset>"
gcs.tmp-bucket = "pg-export"
}

**Comment by ZonerDan** [comment]
Other things I have tried:

- changing gcs.tmp-bucket to gcs.tmpBucket
- changing value of password from """" to nothing
- changing value of password from """" to null

**Comment by ZonerDan** [comment]
Provided in comments

**Comment by sderosiaux** [comment]
I just copy/pasted your HOCON (with all the placeholders <xxx>), the program starts:
```
/bin/pg2bq -Dconfig.file=/Users/sderosiaux/pg2bq.conf
2018-02-20 16:34:04,336  [main] ...
```

**Comment by ZonerDan** [comment]
:( I'm at a loss.  I just tried with all the placeholders as well and still receive the WrongType error.

**Comment by ZonerDan** [comment]
Ok. I think its because in my original testing I received an error stating that I should change the file extension to .properties instead of .conf.  Now I've changed it back and the program is starting.

**Comment by sderosiaux** [comment]
Yeah, I think when the file ends by `.properties`, the config expects to find the Java properties format.
Happy you got it fixed!

**Comment by ZonerDan** [comment]
Thanks for your help!
